### PR TITLE
Configurable Metadata Type

### PR DESCRIPTION
**Pull request by nethad**
In our app, we needed to extend the metadata on collections and didn't find a way to extend the metadata type elegantly.

This PR proposes a change to the `#collection_type` that provides a `metadata_type` argument that is optional. By default, it's the `GraphqlPagination::CollectionMetadataType` but it can be overriden with a custom type that inherits from it to add fields.

Any objections to this change/solution? If it's fine I'd add a section in the README that explains how to extend the metadata, but wanted to make sure that the solution is accepted first.

Thank you!

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ require:
   - rubocop-performance
 
 inherit_from:
-  - https://raw.githubusercontent.com/RenoFi/rubocop/master/0.78.yml
+  - https://raw.githubusercontent.com/RenoFi/rubocop/master/0.80.yml
```
Comment (nethad) [inline .rubocop.yml]: `0.78.yml` didn't exist anymore, so I've changed it to `0.80.yml`, which is the only file available in the repo. With this step, I also needed to upgrade rubocop.

File: graphql-pagination.gemspec [hunk 2]
```diff
@@ -3,24 +3,24 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
...
   spec.metadata['homepage_uri'] = 'https://github.com/RenoFi/graphql-pagination'
   spec.metadata['source_code_uri'] = 'https://github.com/RenoFi/graphql-pagination'
 
   spec.files = Dir.chdir(__dir__) do
     `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(bin/|spec/|\.rub)}) }
   end
-  spec.bindir        = 'exe'
-  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.bindir = 'exe'
+  spec.executables = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
```
Comment (nethad) [inline graphql-pagination.gemspec]: Rubocop auto-correction.

File: lib/graphql_pagination/collection_metadata_type.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
 module GraphqlPagination
   class CollectionMetadataType < GraphQL::Schema::Object
     field :current_page, Integer, null: false
-    field :limit_value,  Integer, null: false
-    field :total_count,  Integer, null: false
-    field :total_pages,  Integer, null: false
+    field :limit_value, Integer, null: false
+    field :total_count, Integer, null: false
+    field :total_pages, Integer, null: false
```
Comment (nethad) [inline lib/graphql_pagination/collection_metadata_type.rb]: Rubocop auto-correction.

**Comment by knapo** [comment]
Thanks @nethad! It definitely makes sense! Sorry about the confusion around rubocop.

**Comment by knapo** [comment]
@nethad I'm merging the PR. Can you propose README changes in a separate one ? Thanks!

**Comment by nethad** [comment]
@knapo That was a speedy review, thank you! :) I'll provide the README change in a follow-up PR. :+1:

**Pull request merged**